ui/src/app/edge/live/common/consumption/modal/modal.constants.spec.ts

```diff
@@ -0,0 +1,12 @@
+import { DummyConfig } from "src/app/shared/edge/edgeconfig.spec";
+import { EdgeConfig } from "src/app/shared/shared";
+import { TestContext } from "src/app/shared/test/utils.spec";
+import { Role } from "src/app/shared/type/role";
+
+import { OeFormlyViewTester } from "../../../../../shared/genericComponents/shared/tester";
+import { ModalComponent } from "./modal";
+
+export function expectView(config: EdgeConfig, role: Role, viewContext: OeFormlyViewTester.Context, testContext: TestContext, view: OeFormlyViewTester.View): void {
+  expect(OeFormlyViewTester.apply(ModalComponent.generateView(DummyConfig.convertDummyEdgeConfigToRealEdgeConfig(config), role, testContext.translate), viewContext))
+    .toEqual(view);
```

Comment:
The functionality and logic are good, but for the sake of readability, I recommend the following approach:

```
const edgeConfig = DummyConfig.convertDummyEdgeConfigToRealEdgeConfig(edgeConfig);
  const generatedView = OeFormlyViewTester.apply(ModalComponent.generateView(edgeConfig, role, testContext.translate), viewContext);

expect(generatedView).toEqual(expectedView);
```